src/aggregator/server/server.go

```diff
@@ -57,6 +57,10 @@ type RunOptions struct {
 	// InterruptCh is a programmatic interrupt channel to supply to
 	// interrupt and shutdown the server.
 	InterruptCh <-chan error
+
+	// ShutdownCh is an optional channel to supply if interested in receiving
+	// a notification that the server has shutdown.
+	ShutdownCh chan<- struct{}
```

Comment:
Why empty struct? Not sure what the convention is, just curious.

Reply:
just idiomatic go. makes it clear it's just a signal and the type doesn't matter.

Reply:
nit: `ShutdownCh` sounds like the channel to actually shutdown. maybe `ShutdownNotifyCh`?